File: services/cookbook/service/src/main/java/family/haschka/wolkenschloss/cookbook/recipe/Servings.java
package family.haschka.wolkenschloss.cookbook.recipe;

import javax.json.bind.annotation.JsonbCreator;
import javax.json.bind.annotation.JsonbProperty;

import static java.lang.String.valueOf;

public record Servings(@JsonbProperty("amount") int amount) {

    public static final int MIN = 1;
    public static final int MAX = 100;

    @JsonbCreator
    public Servings {
        if (amount < MIN || amount > MAX) {
            throw new IllegalArgumentException();
        }
    }

    @Override
    public String toString() {
        return valueOf(amount);
    }
}

File: services/cookbook/core/src/main/kotlin/recipe/Servings.kt
package family.haschka.wolkenschloss.cookbook.recipe

data class Servings(val amount: Int) {
    init {
        if (amount < MIN || amount > MAX) {
            throw IllegalArgumentException()
        }
    }

    fun amount(): Int {
        return amount
    }

    override fun toString(): String {
        return amount.toString()
    }

    companion object {
        const val MIN = 1
        const val MAX = 100
    }
}